src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaCommon.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="KafkaCommon.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using System.Text;
+using OpenTelemetry.AutoInstrumentation.DuckTyping;
+using OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.DuckTypes;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka;
+
+internal static class KafkaCommon
+{
+    public const string ConsumerGroupIdConfigKey = "group.id";
+    private static readonly Type HeadersType;
+
+    static KafkaCommon()
+    {
+        HeadersType = Type.GetType("Confluent.Kafka.Headers, Confluent.Kafka")!;
```

Comment:
I will change it to follow the approach from datadog's [imlementation](https://github.com/DataDog/dd-trace-dotnet/blob/6c051698537075c2c4883bc78ea8f248c8e29465/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/CachedMessageHeadersHelper.cs)

Reply:
Changed to simple implementation based on one linked above, using `Activator.CreateInstance` for now.
If optimized implementation, similar to one from datadog's repository is desired, I'd prefer for that to be done as a part of a separate PR.
Relevant blog post with benchmarks: https://andrewlock.net/benchmarking-4-reflection-methods-for-calling-a-constructor-in-dotnet/#the-results